protocol/client/messenger_v2.go

```diff
@@ -51,32 +51,44 @@ func (m *MessengerV2) Start() error {
 		if err != nil {
 			return err
 		}
+
 		if contacts[i].Type == ContactPublicKey {
 			_, exist := m.private[contacts[i].Topic]
+
 			if exist {
 				continue
 			}
+
 			stream := NewStream(context.Background(), options, m.proto, NewPrivateHandler(m.db))
 			err := stream.Start()
 			if err != nil {
 				return errors.Wrap(err, "unable to start private stream")
 			}
+
 			m.private[contacts[i].Topic] = stream
 		} else {
 			_, exist := m.public[contacts[i].Topic]
+
 			if exist {
 				return fmt.Errorf("multiple public chats with same topic: %s", contacts[i].Topic)
 			}
+
 			stream := NewStream(context.Background(), options, m.proto, NewPublicHandler(contacts[i], m.db))
 			err := stream.Start()
 			if err != nil {
 				return errors.Wrap(err, "unable to start stream")
 			}
+
 			m.public[contacts[i].Topic] = stream
 		}
 	}
-	log.Printf("[INFO] request messages from mail sever")
-	return m.RequestAll(context.Background(), true)
+
+	log.Printf("[Mesenger:Start] request messages from mail sever")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```

Comment:
this timeout is not enough. it usually it  takes 1-2 minutes to download 24h of messages on new accounts

Reply:
I think it might not work then :D I will remove it and add an issue to examine the usage of proper contexts through the whole messenger.